### DISCUSSION TITLE
HW_PWM for ESP32

### DISCUSSION THREAD

**Discussion by pljakobs** [SmingHub/Sming#2576] [General]
as far as I can see, HW_PWM is not currently supported for the ESP32.
Is that something that is in the works for 2.7?
If not, I might go ahead and see if I can contribute.

pj

**Comment by mikee47** [comment]
Hello! As far as I'm aware there's currently no work on that, so please feel free to have a go. The esp32 has very capable hardware PWM (the ledc_xxx stuff) so may warrant a different interface... or an enhanced one at least.
